### ISSUE TITLE
需要改进的地方

### ISSUE THREAD

**Issue by ilaunchmoon** [datawhalechina/self-llm#431]
### 出bug的具体模型

关于Chat-嬛嬛项目下的train.py模块下process_func()方法疑问

### 出bug的具体模型教程

为什么在这个函数中的input_ids和labels都添加[tokenizer.pad_token_id]，不应该是使用[tokenizer.eos_token_id]

### 教程负责人

@ 不要葱姜蒜 KMnO4-zx

### Bug描述

def process_func(example):
    MAX_LENGTH = 384    # Llama分词器会将一个中文字切分为多个token，因此需要放开一些最大长度，保证数据的完整性
    input_ids, attention_mask, labels = [], [], []
    instruction = tokenizer(f"<|begin_of_text|><|start_header_id|>system<|end_header_id|>\n\nCutting Knowledge Date: December 2023\nToday Date: 26 Jul 2024\n\n现在你要扮演皇帝身边的女人--甄嬛<|eot_id|><|start_header_id|>user<|end_header_id|>\n\n{example['instruction'] + example['input']}<|eot_id|><|start_header_id|>assistant<|end_header_id|>\n\n", add_special_tokens=False)  # add_special_tokens 不在开头加 special_tokens
    response = tokenizer(f"{example['output']}<|eot_id|>", add_special_tokens=False)
    input_ids = instruction["input_ids"] + response["input_ids"] + [tokenizer.pad_token_id]
    attention_mask = instruction["attention_mask"] + response["attention_mask"] + [1]  # 因为eos token咱们也是要关注的所以 补充为1
    labels = [-100] * len(instruction["input_ids"]) + response["input_ids"] + [tokenizer.pad_token_id]  
    if len(input_ids) > MAX_LENGTH:  # 做一个截断
        input_ids = input_ids[:MAX_LENGTH]
        attention_mask = attention_mask[:MAX_LENGTH]
        labels = labels[:MAX_LENGTH]
    return {
        "input_ids": input_ids,
        "attention_mask": attention_mask,
        "labels": labels
    }

我的问题是：

为什么在这个函数中的input_ids和labels都添加[tokenizer.pad_token_id]，不应该是使用[tokenizer.eos_token_id]
如果[tokenizer.pad_token_id]起到结束特殊标签的作用，应该使用[tokenizer.eos_token_id]，否则会产生歧义

### 复现步骤

不需要复现

### 期望行为

无

### 环境信息

无

### 其他信息

无

### 确认事项 / Verification

- [x] 此问题未在过往Issue中被报告过 / This issue hasn't been reported before
